tests/test_compiler.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the compiler subpackage.
+"""
+
+import pytest
+
+import pennylane as qml
+
+pytest.importorskip("catalyst")
+pytest.importorskip("jax")
```

Comment:
Here you can mark it with `pytestmark = pytest.mark.external`, it will be run in a runner with catalyst and jax.